tt_metal/impl/device/device.cpp

```diff
@@ -1194,6 +1201,84 @@ CommandQueue& Device::command_queue(size_t cq_id) {
     return *sw_command_queues_[cq_id];
 }
 
+void Device::push_work(std::function<void()> work_executor, bool blocking) {
```

Comment:
It seems like the addition of these new attributes/methods could've been added external WorkerExecutor class and avoid bloating Device class? Is there any reason not to do so?

Reply:
The intention here was to introduce a worker queue and a thread to each device/chip through the device class. This way, the user can launch work for a specific device by doing something like `device->push_work`, rather than looking up the executor linked to a device and then launching work through that.

I guess another way of encapsulating this functionality is to place the worker_thread and worker_queue inside an executor class, that each device can own. All device APIs using the queue can then go through the executor, rather than accessing the raw objects.

Reply:
Yeah, that's what I was thinking.. It should get cleaned up and if it's not a huge overhead to refactor these methods, I think it's worthwhile to do.